cookbooks/swift/recipes/setup.rb

```diff
@@ -54,92 +47,86 @@
   action [:disable, :stop]
 end
 
-# packages
+# system packages
 required_packages = [
   "libssl-dev", # libssl-dev is required for building wheels from the cryptography package in swift.
-  "curl", "gcc", "memcached", "rsync", "sqlite3", "xfsprogs", "git-core", "build-essential",
+  "curl", "gcc", "memcached", "rsync", "sqlite3", "xfsprogs", "git", "build-essential",
   "libffi-dev",  "libxml2-dev", "libxml2", "libxslt1-dev", "zlib1g-dev", "autoconf", "libtool",
   "haproxy", "docker-compose", "rclone",
 ]
 
-if node['platform_version'] == '22.04'
-  required_packages += [
-    "python2-dev", "python2", "python3", "python3-dev",
-    "python3.7", "python3.7-dev", "python3.7-distutils",
-    "python3.8", "python3.8-dev", "python3.8-distutils",
-    "python3.9", "python3.9-dev", "python3.9-distutils",
-  ]
-else
+# common python versions
+required_packages += [
+  # most of time these come from deadsnakes
+  "python3.7", "python3.7-distutils",
+  "python3.8", "python3.8-distutils",
+  "python3.9", "python3.9-distutils",
+  "python3.10",
+  "python3.11",
+  "python3.12",
+  "python3.13",  # edge of technology!
+  # python3 will be redundant with distro version, -dev is needed for pyeclib
+  "python3", "python3-dev",
+]
+
+# only focal has the *really* old py3
+if node['platform_version'].to_i <= 20 then
   required_packages += [
-    "python-dev",
-    "python3.6", "python3.6-dev", "python3.7", "python3.7-dev",
-    "python3.8", "python3.8-dev",
+    "python3.6", "python3.6-distutils",
   ]
 end
 
-extra_packages = node['extra_packages']
-(required_packages + extra_packages).each do |pkg|
-  package pkg do
-    action :install
-  end
-end
-
-# no-no packages (PIP is the bomb, system packages are OLD SKOOL)
+# no-no packages (PIP rules this vm, most system packages are all out-of-date anyway)
 unrequired_packages = [
-  "python-requests",  "python-six", "python-urllib3",
-  "python-pbr", "python-pip",
-  "python3-requests",  "python3-six", "python3-urllib3",
-  "python3-pbr", "python3-pip",
+  "python3-pip", "python3-pbr", "python3-setuptools",
+  "python3-openssl", "python3-certifi",
+  "python3-requests",  "python3-urllib3",
 ]
 unrequired_packages.each do |pkg|
   package pkg do
     action :purge
   end
 end
 
-if node['use_python3']
-  default_python = 'python3'
-  pip_url = 'https://bootstrap.pypa.io/get-pip.py'
-else
-  default_python = 'python2'
-  pip_url = 'https://bootstrap.pypa.io/pip/2.7/get-pip.py'
+# good-good packages (do the install after purge)
+extra_packages = node['extra_packages']
+(required_packages + extra_packages).each do |pkg|
+  package pkg do
+    action :install
+  end
 end
 
-execute "select default python version" do
-  command "ln -sf #{default_python} /usr/bin/python"
+# see https://peps.python.org/pep-0668/
+file 'break system python' do
+  path "/etc/pip.conf"
+  content <<-EOF
+[global]
+root-user-action = ignore
+break-system-packages = true
```

Comment:
From https://github.com/NVIDIA/vagrant-swift-all-in-one/pull/156 I thought you *didn't* like this approach -- did it turn out it *was* "inexorably linked to supporting noble"?

Reply:
> inexorably linked to supporting noble
IMHO, yes it's "inexorably linked to supporting `sudo pip`" !